src/main/java/com/iota/iri/utils/log/interval/IntervalLogger.java

```diff
@@ -304,17 +315,25 @@ public int hashCode() {
          */
         @Override
         public boolean equals(Object obj) {
-{            if (obj == this) {
+            if (obj == this) {
                 return true;
             }
 
-            if (!getClass().equals(obj.getClass())) {
+            if (obj == null || !getClass().equals(obj.getClass())) {
                 return false;
             }
 
             return message.equals(((Message) obj).message);
```

Comment:
Just a question are we sure `message` will never be null?
Otherwise equals may throw null pointer exception.

I'd opt to do: `Objects.equals(message, ((Message) obj).message)`